### PR TITLE
added MailFolder, moved trash,inbox,sent screens to mailFolderScreen

### LINKED ISSUE #1044: consider all folders screens to use the same class (or base) in ui tests
Inbox, sent, trash, and other such screens are all the same - they are all just a folder screen. So it should probably really just be one screen. A `MailFolderScreen` or maybe just `InboxScreen` or if you want to extend from something and still have them all separate, then `BaseFolderScreen`. What do you think?

_Originally posted by @tomholub in https://github.com/FlowCrypt/flowcrypt-ios/pull/1042#discussion_r753275476_

### PR DESCRIPTION
**Pull request by fcvakintos**
This PR contains new mailFolderScreen, which has methods from trash,inbox,sent screens, removed trash/inbox/sent srcreens

close #1044 

----------------------------------

**Tests** 
- Tests added or updated

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [x] addresses the issue it closes (if any)
- [x] code is readable and understandable
- [x] is accompanied with tests, or tests are not needed
- [x] is free of vulnerabilities
- [x] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

File: appium/tests/screenobjects/mail-folder.screen.ts [hunk 1]
```diff
@@ -1,31 +1,63 @@
...
 
 const SELECTORS = {
-  ENTER_YOUR_PASS_PHRASE_FIELD: '-ios class chain:**/XCUIElementTypeSecureTextField[`value == "Enter your pass phrase"`]',
-  OK_BUTTON: '~Ok',
-  CONFIRM_PASS_PHRASE_FIELD: '~textField',
+  TRASH_HEADER: '-ios class chain:**/XCUIElementTypeNavigationBar[`name == "TRASH"`]',
+  SENT_HEADER: '-ios class chain:**/XCUIElementTypeNavigationBar[`name == "SENT"`]',
   CREATE_EMAIL_BUTTON: '-ios class chain:**/XCUIElementTypeButton[`label == "+"`]',
-  INBOX_HEADER: '-ios class chain:**/XCUIElementTypeStaticText[`label == "INBOX"`]'
+  INBOX_HEADER: '-ios class chain:**/XCUIElementTypeStaticText[`label == "INBOX"`]',
```
Comment (tomholub) [inline appium/tests/screenobjects/mail-folder.screen.ts:9]: Let's use accessibility identifiers for these.

Wouldn't the tests all break if one email subject had "TRASH" or "+" in it?

**Review by tomholub — CHANGES_REQUESTED** [review]
Looks good - less duplication.

When you encounter any `ios class chain` kind of selector, use accessibility identifiers instead.

File: appium/tests/screenobjects/mail-folder.screen.ts (hunk 1 above)
Comment (fcvakintos) [reply appium/tests/screenobjects/mail-folder.screen.ts:9]: @tomholub I created new issue for this

**Review by tomholub — DISMISSED** [review]
I can merge it since it was filed as a separate issue, but merge conflicts need to be resolved first.

**Comment by fcvakintos** [comment]
resolved

**Comment by fcvakintos** [comment]
@tomholub can you merge it

**Review by tomholub — APPROVED** [review]

**Pull request merged**